SiPixelPayloadInspectorHelper: warning: '<anonymous>' may be used uninitialized

A warning is issued in DBG_X IB CMSSW_13_1_DBG_X_2023-03-02-2300:
```
In member function 'filterThePartition',
    inlined from 'attachLocationLabel.constprop' at .../src/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc:933:97:
  .../src/CondCore/SiPixelPlugins/interface/SiPixelPayloadInspectorHelper.h:772:15: warning: '<anonymous>' may be used uninitialized [-Wmaybe-uninitialized]
   772 |           if (*m_Phase < SiPixelPI::phase::two) {
      |               ^
.../src/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc: In member function 'attachLocationLabel.constprop':
.../src/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc:932:40: note: '<anonymous>' declared here
  932 |         t_info_fromXML.fillGeometryInfo(detid, tTopo, phInfo.phase());
      |                                        ^
```

Full log: [link](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc11/CMSSW_13_1_DBG_X_2023-03-02-2300/CondCore/SiPixelPlugins)

--- a/CondCore/SiPixelPlugins/interface/SiPixelPayloadInspectorHelper.h
+++ b/CondCore/SiPixelPlugins/interface/SiPixelPayloadInspectorHelper.h
@@ -36,7 +36,7 @@
 
 namespace SiPixelPI {
 
-  enum phase { zero = 0, one = 1, two = 2 };
+  enum phase { zero = 0, one = 1, two = 2, undefined = 999 };
 
   // size of the phase-0 pixel detID list
   static const unsigned int phase0size = 1440;
@@ -664,7 +664,7 @@ namespace SiPixelPI {
     int m_side;
     int m_ring;
     bool m_isInternal;
-    SiPixelPI::phase* m_Phase;
+    SiPixelPI::phase m_Phase;
 
   public:
     void init();
@@ -687,6 +687,7 @@ namespace SiPixelPI {
   inline void topolInfo::init()
   /*--------------------------------------------------------------------*/
   {
+    m_Phase = SiPixelPI::undefined;
     m_rawid = 0;
     m_subdetid = -1;
     m_layer = -1;
@@ -710,7 +711,7 @@ namespace SiPixelPI {
   /*--------------------------------------------------------------------*/
   {
     // set the phase
-    m_Phase = const_cast<SiPixelPI::phase*>(&ph);
+    m_Phase = ph;
     unsigned int subdetId = static_cast<unsigned int>(detId.subdetId());
 
     m_rawid = detId.rawId();
@@ -733,7 +734,7 @@ namespace SiPixelPI {
   {
     SiPixelPI::regions ret = SiPixelPI::NUM_OF_REGIONS;
 
-    if (m_Phase == nullptr) {
+    if (m_Phase == SiPixelPI::undefined) {
       throw cms::Exception("LogicError") << "Cannot call filterThePartition BEFORE filling the geometry info!";
     }
 
@@ -769,7 +770,7 @@ namespace SiPixelPI {
           m_side > 1 ? ret = SiPixelPI::FPixpL3 : ret = SiPixelPI::FPixmL3;
           break;
         default:
-          if (*m_Phase < SiPixelPI::phase::two) {
+          if (m_Phase < SiPixelPI::phase::two) {
             // warning message only if the phase2 is < 2
             edm::LogWarning("LogicError") << "Unknow FPix disk: " << m_layer;
           }

--- a/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc
+++ b/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc
@@ -197,6 +197,8 @@ namespace {
         case SiPixelPI::phase::two:
           inputFile = "Geometry/TrackerCommonData/data/PhaseII/trackerParameters.xml";
           break;
+        default:
+          throw cms::Exception("SiPixelDynamicInefficiency_PayloadInspector") << "checkPhase: unrecongnized phase!";
       }
 
       // create the standalone tracker topology

